imap/sync_support.c

```diff
@@ -2162,7 +2205,7 @@ int sync_append_copyfile(struct mailbox *mailbox,
 
     if (!item || !item->fname)
         r = IMAP_IOERROR;
-    else
+    else if (!record->cache_version)
```

Comment:
This seems an unrelated optimisation to be in this commit!

Reply:
It is related in that sync_sieve_validate() has parsed the message during the first pass, so we avoid parsing it a second time